app/services/twilio/incoming_message_service.rb

```diff
@@ -19,22 +21,25 @@ def perform
 
   private
 
-  def twilio_inbox
-    @twilio_inbox ||=
-      ::Channel::TwilioSms.find_by(messaging_service_sid: params[:MessagingServiceSid]) ||
-      ::Channel::TwilioSms.find_by!(account_sid: params[:AccountSid], phone_number: params[:To])
+  def twilio_channel
+    @twilio_channel ||= ::Channel::TwilioSms.find_by(messaging_service_sid: params[:MessagingServiceSid]) if params[:MessagingServiceSid].present?
+    if params[:AccountSid].present? && params[:To].present?
```

Comment:
@tejaswinichile lets refactor this later
 looking further into this as an if else case. one could have message service id assigned for a number. but decided to configure the chatwoot inbox with phone number /account sid configuration.  ( so in params MessageServiceSid could be present ) ( even though the channel is configured based on accountSid and Number

Since existing code mirrors that behaviour. lets leave it like that and revisit.